thoth/common/workflows.py

```diff
@@ -480,7 +484,14 @@ def submit_workflow_from_template(
         :param template_parameters: parameters for the template
         :param workflow_parameters: parameters for the workflow
         :param workflow_namespace: namespace to submit the workflow to
+        :param workflow_limit: limit number of workflows currently in memory for workflowController
         """
+        if workflow_limit is not None:
+            if workflow_namespace is None:
+                raise ValueError("Namespace must be set when passing a limit.")
+            while self._get_total_workflows(workflow_namespace) > workflow_limit:
+                _LOGGER.debug("Waiting for number of workflows to drop below limit.")
```

Comment:
Can this be done on the library user side? It feels not that nice to do it on the library side.

Also, the active wait will issue a lot of requests, some sleeping would be appropriate in this case.

Reply:
Yeah, a sleep would be good here.  I thought it would be nice to implement here, otherwise we will be duplicating the functionality a lot.

Reply:
> otherwise we will be duplicating the functionality a lot.

How many components would call this method?

Reply:
It would be for any component which is a `faust` consumer. Which includes at least investigator, package-update-job, and package-release

Reply:
> package-update-job, and package-release

I think all the components should produce messages which will be handled by investigator - so I think only investigator will schedule workflows.

Reply:
Okay. I'll create an issue in investigator indicating this so that we know that we should be moving the logic over there.